server/sonar-server/src/main/java/org/sonar/server/permission/ws/template/DefaultTemplatesResolver.java

```diff
@@ -17,20 +17,35 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-package org.sonar.server.permission;
+package org.sonar.server.permission.ws.template;
 
-import static java.lang.String.format;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.Immutable;
+import org.sonar.db.organization.DefaultTemplates;
 
-public class DefaultPermissionTemplates {
-  public static final String DEFAULT_TEMPLATE_PROPERTY = "sonar.permission.template.default";
-  public static final String DEFAULT_TEMPLATE_KEY = "default_template";
-  private static final String DEFAULT_ROOT_QUALIFIER_TEMPLATE_PATTERN = "sonar.permission.template.%s.default";
+import static java.util.Objects.requireNonNull;
+import static java.util.Optional.ofNullable;
 
-  private DefaultPermissionTemplates() {
-    // utility class
-  }
+public interface DefaultTemplatesResolver {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Annotate the "DefaultTemplatesResolver" interface with the @FunctionalInterface annotation [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1609)